### ISSUE TITLE
Dependencies of filetype-specific plugins should be filetype-specific too

### ISSUE THREAD

**Issue by p00f** [wbthomason/packer.nvim#139]
I specified `clojure-vim/vim-jackin` like
```lua
'clojure-vim/vim-jack-in',
    requires = {
      'radenling/vim-dispatch-neovim',
      'tpope/vim-dispatch'
    },
    ft = 'clojure'
```
but `vim-jack-in` is installed in opt while while `vim-dispathc-neovim` and `vim-dispatch` are installed in `start`

**Comment by wbthomason** [comment]
Good point. Right now, parent config values don't apply to requirements, but I suppose it makes sense for some of them to. Might need to think about this in the context of #87 and sequenced loads...

**Comment by p00f** [comment]
> #87

oh right
